### PR TITLE
Remove /std/media_types

### LINKED ISSUE #4292: media_types API is not good
The media_types API employs too much magic leading to unexpected behavior:

```
lookup('html');                     // 'text/html'
lookup('file.html');                // 'text/html'
lookup('path/to/file.html');        // 'text/html'

contentType('html');                // 'text/html; charset=utf-8'
contentType('file.html');           // 'text/html; charset=utf-8'
contentType('path/to/file.html');   // 'path/to/file.html' - uhh, what??
```

Also, the naming, while understandable for someone who knows the implementation (e.g. how the database is keyed), is a bit confusing for a newcomer:
  - `lookup()`? Aren't they all lookups of some type or another?
  - `lookup()` returns a (non-full?) "content type", which you pass to `contentType()` and it returns a "full" content type" ?

So many exports/functions:
  - `lookup(path)`
  - `contentType(type)`
  - `extension(type)`
  - `charset(type)`
  - `extensions`
  - `types`

A (imo) much easier to understand API would be to just expose (maybe a subset of) the database record type:

```
export interface MediaType {
    identifier: string,
    extensions: string[],
    charset?: string,
    compressible?: boolean
}

export function lookupByIdentifier(identifier: string): MediaType|undefined
export function lookupByExtension(path: string): MediaType|undefined

```

### PR DESCRIPTION
**Pull request by ry**
Fix #4292

### CODE REVIEW

**Comment by kitsonk** [comment]
I will create another repo today and raise a PR for denoland/x.

Just a thought, `std/http/file_server.ts` is really just an example.  It doesn't export anything and it really isn't that super useful on its own.  It isn't a library, as it doesn't export anything.  Maybe we should earmark that a bit better its intent.

**Comment by kitsonk** [comment]
Moved it to oakserver and created a PR here: denoland/deno_website2#299

File: std/http/file_server.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ const CORSEnabled = serverArgs.cors ? true : false;
 const target = posix.resolve(serverArgs._[1] ?? "");
 const addr = `0.0.0.0:${serverArgs.port ?? serverArgs.p ?? 4500}`;
 
-const MEDIA_TYPES = {
+const MEDIA_TYPES: { [key: string]: string } = {
```
Comment (kitsonk) [inline std/http/file_server.ts:43]:
```suggestion
const MEDIA_TYPES: Record<string, string> = {
```

**Comment by ry** [comment]
Moved to https://deno.land/x/media_types/

File: std/http/file_server.ts [hunk 2]
```diff
@@ -41,6 +40,25 @@ const CORSEnabled = serverArgs.cors ? true : false;
...
 const addr = `0.0.0.0:${serverArgs.port ?? serverArgs.p ?? 4500}`;
 
+const MEDIA_TYPES: Record<string, string> = {
+  ".md": "text/markdown",
+  ".html": "text/html",
+  ".htm": "text/html",
+  ".json": "application/json",
+  ".map": "application/json",
+  ".txt": "text/plain",
+  ".ts": "application/typescript",
```
Comment (zhmushan) [inline std/http/file_server.ts:50]: @ry I just tried it and found that using `application/typescript`, the browser will download the file directly, I think that `.ts` should be mapped to "application/javascript" or "application/typescript; charset = UTF-8"

**Pull request merged**